### DISCUSSION TITLE
Add staking to the KRE

### DISCUSSION THREAD

**Discussion by emartin59** [kinfoundation/kin-rfcs#166] [KRE 4.0 Discussion]
I recently thought of a way to add a staking element for the developers who are paid out for the KRE. The simple explanation is that they automatically get some payout, but they can get more by leaving Kin in a designated wallet.

Here it is on Reddit, a bit of traction and Ted seems to think it has potential:

https://www.reddit.com/r/KinFoundation/comments/wxeolk/staking_and_the_kre/

I just tried adding information from my Reddit idea into the rewards engine files on GitHub. There's a little more detail here:

https://github.com/kinecosystem/rewards-engine/compare/master...emartin59:rewards-engine:patch-1

Thanks!

ideaDash / Eric

**Comment by hann228** [comment]
Thanks for your proposal.  We have looked backed through historical proposals and think this is unique in terms of its proposed design.   

We had some questions for you:

Developers may distribute Kin from their wallet for reasons other than selling - for example many Kin apps today distribute large amounts of Kin from these wallets to their user base.  The KRE 3 live today is based on distributing the most Kin to users as possible.  For example apps like PeerBet share the bulk of the Kin received from the KRE with their user base - other apps which have implemented loyalty or staking type rewards have broadly similar designs.  Question here is - if the KRE payments are significantly based around the amount of Kin held in the developer wallet - wouldn't this simply act as a penalty for rewarding users or at least lead to the reduction of the amount of Kin distributed to the user economy.  How do you see this working with your proposal?

How would this proposal deal with the runaway train scenario?  That is, a developer who starts by staking a larger amount, will in turn receive a larger reward, which in turn will lead to another large reward the next week - making it difficult for others to catch up.

**Comment by emartin59** [comment]
All issues to consider. Another proposed a separate staking benefit
regardless of whether someone is a developer. I like that. Also, maybe
there's a way we can encourage giving away and staking half, or separate
staking. Not sure! Something's gotta give because Kin's value has been
languishing. Something's not working. Maybe general staking is the fix, or
a bit of both. Encourage some staking from developers, some from investors,
and encourage some KRE reward give awayage by developers. Not sure!

**Comment by hann228** [comment]
Ok, what we can do here is add this to our KRE research stream and keep looking at it. 

If you have any more ideas on how this might work then keep posting them to this thread.

**Comment by emartin59** [comment]
Great, thanks!
